### PR TITLE
Document/ivyportal 16663 quick search in dashboard task list widget lts documentation

### PR DESCRIPTION
**Pull request by lmluat-axonivy**

### CODE REVIEW

File: Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst [hunk 1]
```diff
@@ -336,3 +339,60 @@ Below is the list of filterable columns and their corresponding filter condition
...
+            ...
+         }
+
+      ..
+
+      Valid values:
+
+      * ``true``: show the quick search text box.
+      * ``false``: hide the quick search text box.
+      * ``not define``: hide the quick search text box.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:368]: not define --> not defined

File: Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst [hunk 2]
```diff
@@ -336,3 +339,60 @@ Below is the list of filterable columns and their corresponding filter condition
...
+
+   * ``quickSearch``: to choose which columns can be search by the quick search
+     feature, set the ``quickSearch`` field for each column as below.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:371]: as below --> as shown below

File: Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst [hunk 3]
```diff
@@ -336,3 +339,60 @@ Below is the list of filterable columns and their corresponding filter condition
...
+
+      ..
+
+      Valid values:
+
+      * ``true``: show the quick search text box.
+      * ``false``: hide the quick search text box.
+      * ``not define``: hide the quick search text box.
+
+   * ``quickSearch``: to choose which columns can be search by the quick search
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:370]: can be search --> can be searched

File: Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst [hunk 4]
```diff
@@ -336,3 +339,60 @@ Below is the list of filterable columns and their corresponding filter condition
       the task widget will show tasks that have expiry dates between the dates defined.
 
       Acceptable date formats: ``dd.MM.yyyy`` and ``MM/dd/yyyy``.
+
+Quick Search
+------------
+
+Quick search is a useful function for user to search quickly on the task widget. 
+
+   * ``enableQuickSearch``: to enable/disable the quick search feature, set the
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:348]: I see that you have two locations for the enableQuickSearch attribute: on widget level and on column level.

What happens if I enable the widget 

- on widget level only
- on column level, but not on widget level
- on column level and on widget level?

It is important to know if you can disable the feature by not enabling it on the widget level - or not.

Please include this expanation right after the initial sentence "Quick search is ...",

File: Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst [hunk 5]
```diff
@@ -103,9 +103,20 @@ In the column management panel, you will find two sections:
...
+
+    Quick search feature is supporting these fields:
+
+      #. Standard fields: Id, name, description, category, responsible (display name), and application.
+      #. Task custom fields: fields that have type ``STRING`` or ``TEXT``.
+      #. Case custom fields: fields that have type ``STRING`` or ``TEXT``.
+
+    Quick search is enabled for the standard field name and description by default.
+
 Portal supports to display custom case fields within the task list.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst:120]: supports to display --> supports the display of

File: Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst [hunk 6]
```diff
@@ -119,6 +130,19 @@ You can export all data of the task widget by clicking on the link ``Export to E
 If the number of exported tasks exceed the maximum row number of the Excel file, Portal will separate data into multiple Excel
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst:130]: number is singular, thus:  

exceed --> exceeds

File: Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst [hunk 7]
```diff
@@ -119,6 +130,19 @@ You can export all data of the task widget by clicking on the link ``Export to E
...
 files and put them into a single zip file.
 
+**Quick search**
+
+|quick-search-column-management|
+
+In the view mode, when quick search is enabled, a text box will appear to allow you to search.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst:137]: In the view --> In view
allow you to search --> allow searching

File: Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst [hunk 8]
```diff
@@ -119,6 +130,19 @@ You can export all data of the task widget by clicking on the link ``Export to E
...
+
+|quick-search-checkbox|
+
+  .. important::
+
+    #. Quick search function is unaffected by widget filters.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst:143]: Quick --> The quick

File: Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst [hunk 9]
```diff
@@ -119,6 +130,19 @@ You can export all data of the task widget by clicking on the link ``Export to E
...
+    #. The scope of the quick search feature can be defined in the configuration panel.
```
Comment (phochstr) [inline Documentation/portal-guide/source/portal-user-guide/new-dashboard/index.rst:144]: Do we need to document what scopes are available for selection here?

Avoid passive voice (can be configured), use active voice!
Thus, 
The scope of the quick search feature can be defined in the configuration panel.
-->
You can define the scope of the quick search feature in the configuration panel.

**Review by phochstr — APPROVED** [review]
Adapt and merge.

Good work!

File: Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst (hunk 4 above)
Comment (lmluat-axonivy) [reply Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:363]: Hi @phochstr thank for your feedback, I updated the explanation for this part, please take a look

Comment (phochstr) [reply Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-task-widget.rst:363]: Very good, that makes usage transparent!

**Review by phochstr — APPROVED** [review]
Yes, that's it! 

Thanks for your update.

**Pull request merged**